src/jit/morph.cpp

```diff
@@ -6940,263 +6945,236 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
     //
     // Note that callee being a vararg method is not a problem since we can account the params being passed.
     unsigned nCallerArgs = info.compArgsCount;
+    unsigned nCalleeArgs = argInfo->ArgCount();
 
     size_t callerArgRegCount      = codeGen->intRegState.rsCalleeRegArgCount;
     size_t callerFloatArgRegCount = codeGen->floatRegState.rsCalleeRegArgCount;
 
     // Count the callee args including implicit and hidden.
     // Note that GenericContext and VarargCookie are added by importer while
     // importing the call to gtCallArgs list along with explicit user args.
-    size_t calleeArgRegCount      = 0;
+    size_t callerIntArgRegCount      = 0;
```

Comment:
Shouldn't this be `callee`, not `caller`?

Also, the above comment "Count the callee args including implicit and hidden" should be updated, shouldn't it, since we counted callee args with `nCalleeArgs` above.